### ISSUE TITLE
Reorg and difficulty adjustment error

### ISSUE THREAD

**Issue by adiabat** [mit-dci/lit#119]
There is some kind of interaction between reorgs and difficulty adjustment which causes uspv to think a difficutly change is invalid.  This was on testnet, which I still don't 100% understand the difficulty nerfing algo for, so may be related to that, or may be something simpler.

errors here:
```
2017/11/30 13:18:54.471755 got inv.  Contains:
2017/11/30 13:18:54.471789 	0)MSG_BLOCK : 0000000000000bfccbcc411f787296eb13fddbe4f32858ad3c37885088636711
2017/11/30 13:18:54.471805 asking for headers due to inv block
2017/11/30 13:18:54.471845 got header tip height 1251344
2017/11/30 13:18:54.472130 get headers message has 19 header hashes, first one is 00000000000007eb00fc245c0808b2c04c4d17c3cffaff515944fba61d27fd67
2017/11/30 13:18:54.473163 got 2 headers. Range:
0000000000000da79fd9445b794a6d9f52d834e6eb7c28c559b9fc8985380f3c - 0000000000000bfccbcc411f787296eb13fddbe4f32858ad3c37885088636711
2017/11/30 13:18:54.473212 header file position: 17532160
2017/11/30 13:18:54.473230 made 2017 header slice
2017/11/30 13:18:54.485187 Header 0000000000000da79fd9445b794a6d9f52d834e6eb7c28c559b9fc8985380f3c attaches at height 1251343
2017/11/30 13:18:54.485201 reorg from height 1251344 to 1251345
2017/11/30 13:18:54.485219 Header error: Block 1251345 0000000000000da79fd9445b794a6d9f52d834e6eb7c28c559b9fc8985380f3c incorrect difficuly.  Read 1a3fffc0, expect 1b00ffff
```

**Comment by adiabat** [comment]
another, same situation, 1 orphan block reorg, expects `1b00ffff` (not `1d00ffff` ... hmm)

```2017/11/30 20:54:50.714299 got inv.  Contains:
2017/11/30 20:54:50.714351 	0)MSG_BLOCK : 000000000000224388f3c584f5d0d3a3b0e39cf87cfc0e932e155bcd33debd06
2017/11/30 20:54:50.714375 asking for headers due to inv block
2017/11/30 20:54:50.714413 got header tip height 1251688
2017/11/30 20:54:50.714985 get headers message has 19 header hashes, first one is 00000000000007cf44bf17c7f423eaf88f6952be2b71807647c1001ca480c67b
2017/11/30 20:54:50.715968 got 4 headers. Range:
00000000000007bac7f083a3b691ad640a6d5a40547c3905832d1682b760e759 - 000000000000224388f3c584f5d0d3a3b0e39cf87cfc0e932e155bcd33debd06
2017/11/30 20:54:50.716135 header file position: 17559680
2017/11/30 20:54:50.716173 made 2017 header slice
2017/11/30 20:54:50.735667 Header 00000000000007bac7f083a3b691ad640a6d5a40547c3905832d1682b760e759 attaches at height 1251685
2017/11/30 20:54:50.735685 reorg from height 1251688 to 1251689
2017/11/30 20:54:50.735700 Header error: Block 1251689 00000000000007bac7f083a3b691ad640a6d5a40547c3905832d1682b760e759 incorrect difficuly.  Read 1a3fffc0, expect 1b00ffff
```

**Comment by Varunram** [comment]
Hm, I'm not able to reproduce the bug. On my side, everything's cool (I didn't even get an inv block)
```
2017/12/01 10:05:46.729697 ingested full block 0000000000000775890a8230102288697ea5430ec69d85ab572b5ab79980f9ac height 1251688 OK
2017/12/01 10:05:46.730299 ingested full block 000000000000224388f3c584f5d0d3a3b0e39cf87cfc0e932e155bcd33debd06 height 1251689 OK
```
I'll dig deeper into this and find if there's something else wrong.

**Comment by Varunram** [comment]
Okay, I think I got the bug, its related to the testnet difficulty algo. It happens during the DiffCalcFunction called over at uspv. Weird that we didn't catch it sooner though. I'll update you about it today.

**Comment by adiabat** [comment]
Merged #121 which might fix this but I'm not sure.  Will leave open for a while and eventually close it if it doesn't happen again.

**Comment by adiabat** [comment]
Probably ok as no reports of this
